### PR TITLE
add _DELIMITER_ to split env and uselessheader

### PR DESCRIPTION
**Pull request by saltfish666**
sorry for last bad pull request, I add _DELIMITER_ to split env and useless header for your last comment.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -3,16 +3,18 @@ const execa = require('execa');
...
+	env = env.split('_SHELL_ENV_DELIMITER_')[1];
 	const ret = {};
 
 	for (const line of stripAnsi(env).split('\n')) {
-		const parts = line.split('=');
-		ret[parts.shift()] = parts.join('=');
+		if (line !== '') {
+			const parts = line.split('=');
+			ret[parts.shift()] = parts.join('=');
+		}
```
Comment (silverwind) [inline index.js]: Suggestion:

````js
 	for (const line of stripAnsi(env).split('\n').filter(line => !!line)) {
````

File: index.js [hunk 2]
```diff
@@ -3,16 +3,18 @@ const execa = require('execa');
...
 	}
-
```
Comment (silverwind) [inline index.js]: Keep this line

**Comment by silverwind** [comment]
LGTM once you resolve above comments.

File: index.js (hunk 1 above)
Comment (saltfish666) [reply index.js]: Hi, I just test this code, but after` npm test `, it throws:
```
index.js:12:63
  ✖  12:63  use Boolean(line) instead.  no-implicit-coercion
```
so how about :
```
for (const line of stripAnsi(env).split('\n').filter(line => Bealoon(line))) {
```

Comment (silverwind) [reply index.js]: Ah, yes. `Boolean(line)` instead of `!!line`.

**Pull request merged**